cpp/oneapi/dal/test/engine/csr_table_builder.hpp

```diff
@@ -81,6 +123,29 @@ csr_table copy_data_to_csr(sycl::queue& queue,
                            column_count,
                            indexing);
 }
+
+template <typename Float = float>
+homogen_table copy_data_to_dense(sycl::queue& queue,
+                                 const dal::array<Float>& data,
+                                 const dal::array<std::int64_t>& column_indices,
+                                 const dal::array<std::int64_t>& row_offsets,
+                                 const sparse_indexing indexing,
+                                 const std::int64_t column_count,
+                                 const std::int64_t row_count) {
+    auto dense_data_host = copy_data_to_dense_array(data,
+                                                    column_indices,
+                                                    row_offsets,
+                                                    indexing,
+                                                    column_count,
+                                                    row_count);
+    const auto dense_data_device =
+        dal::array<Float>::empty(queue, row_count * column_count, sycl::usm::alloc::device);
+    auto data_event = queue.copy<Float>(dense_data_host.get_data(),
```

Comment:
It can be simplified with 
`queue.copy<Float>(dense_data_host.get_data(),
dense_data_device.get_mutable_data(),
                                        row_count * column_count).wait_and_throw();`
 And can you please check as much as possible scenarios, because of bug with current dense copy from host to device https://github.com/oneapi-src/oneDAL/pull/2966